### PR TITLE
Migrate document-submit to doc scope

### LINKED ISSUE #5399: Migrate document-submit module to ampdoc scope
See `getServiceForDoc` and family.

### PR DESCRIPTION
**Pull request by mkhatib**
@dvoytenko not sure if this the right thing to do or follow the `document-click` footstep in creating a class for it. I remember during initial review of `document-submit` we didn't want to do that but let me know if that changed with doc-scoped services.

Fixes #5399

### CODE REVIEW

File: src/document-submit.js [hunk 1]
```diff
@@ -25,14 +26,18 @@ const PROP = '__AMP_SUBMIT';
...
  */
-export function installGlobalSubmitListener(window) {
-  if (!window[PROP]) {
-    window[PROP] = true;
-    window.document.documentElement.addEventListener(
-        'submit', onDocumentFormSubmit_, true);
-  }
+export function installGlobalSubmitListenerForDoc(ampdoc) {
+  return /** @type {!Object} */ (getServiceForDoc(ampdoc, 'submit', ampdoc => {
+    if (!ampdoc.win[PROP]) {
```
Comment (dvoytenko) [inline src/document-submit.js]: Well, the code here didn't really change :)

What we need is:
1. Track whether the service is installed on ampdoc level, not window. I.e. there will be as many listeners as there ampdocs.
2. The listener should be installed on ampdoc's root node.
3. Why do we need `win[PROP]` here? `getServiceForDoc` already tests the existence.

Comment (mkhatib) [inline src/document-submit.js]: Done

**Comment by mkhatib** [comment]
PTAL 👀

File: src/document-submit.js [hunk 2]
```diff
@@ -14,25 +14,23 @@
...
 
 
-/** @const {string} */
-const PROP = '__AMP_SUBMIT';
-
-
 /**
- * @param {!Window} window
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {!Object}
```
Comment (dvoytenko) [inline src/document-submit.js]: Let's remove `return` type here and typecast below.

**Comment by dvoytenko** [comment]
LGTM with one more request.

File: src/document-submit.js (hunk 2 above)
Comment (mkhatib) [inline src/document-submit.js]: Done

**Pull request merged**